### PR TITLE
Apply PHP 7.4 syntax and typed property

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 7.4, php 7.4 syntax can be applied with typed properties. 

- For `final class`, typed properties for protected is allowed if no inherit of parent or used by its final class parent.
- Otherwise, update only private properties.

### CODE REVIEW

File: src/functions/marshal_uri_from_sapi.php [hunk 1]
```diff
@@ -138,45 +126,39 @@ function marshalUriFromSapi(array $server, array $headers): Uri
...
     $uri = new Uri('');
 
     // URI scheme
     $scheme            = 'http';
-    $marshalHttpsValue = function ($https): bool {
+    $marshalHttpsValue = static function ($https): bool {
         if (is_bool($https)) {
             return $https;
         }
-
```
Comment (Ocramius) [inline src/functions/marshal_uri_from_sapi.php]: Not sure why your CS adjustments removed all these spaces: seems invalid?

Makes the entire thing much less readable too...

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/functions/marshal_uri_from_sapi.php (hunk 1 above)
Comment (samsonasik) [reply src/functions/marshal_uri_from_sapi.php]: Yes, that's known drawback of using rector https://github.com/rectorphp/rector#known-drawbacks , I can rollback the lines if needed.

Comment (Ocramius) [reply src/functions/marshal_uri_from_sapi.php]: Yes please: I know it's annoying, but some of these spacing changes do need adjustment indeed :)

Comment (samsonasik) [reply src/functions/marshal_uri_from_sapi.php]: I rolled back spaces https://github.com/laminas/laminas-diactoros/pull/115/commits/cf8ec668d3f7c189fb578a32992f488914b3603a

**Comment by samsonasik** [comment]
@Ocramius All green 🎉 it is ready now.

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**